### PR TITLE
Allow a default message to be provided for a given pattern.

### PR DESCRIPTION
**Pull request by jamestalmage**
See: https://github.com/sindresorhus/ava/pull/340#issuecomment-167479840

### CODE REVIEW

**Comment by jamestalmage** [comment]
This also cleans up the way we are passing arguments around a bit by changing the signature of `concreteAssert` to also take the original `callSpec` as an argument. This reduces quite a bit of property duplication.

**Comment by twada** [comment]
@jamestalmage Thanks! And sorry for late response.

Is there any place to be changed in README?

**Comment by jamestalmage** [comment]
Yeah, needs some docs.

I am wondering if we shouldn't also just attach their entire original specification to the event, instead of copying over individual properties. That way they can attach arbitrary data specific to their own handlers

**Comment by twada** [comment]
> I am wondering if we shouldn't also just attach their entire original specification to the event, instead of copying over individual properties.

That's what you've done in this pull-req, right?

File: lib/decorator.js [hunk 1]
```diff
@@ -29,10 +30,11 @@ Decorator.prototype.enhancement = function () {
         var callSpec = {
             thisObj: that.receiver,
             func: that.receiver[methodName],
-            numArgsToCapture: numberOfArgumentsToCapture(matcher),
+            numArgsToCapture: numberOfArgumentsToCapture(matcherSpec),
+            matcherSpec: matcherSpec,
```
Comment (jamestalmage) [inline lib/decorator.js:35]: I've attached it here.

File: lib/decorator.js [hunk 2]
```diff
@@ -61,27 +63,30 @@ Decorator.prototype.container = function () {
...
         message = this.config.modifyMessageBeforeAssert({originalMessage: message, powerAssertContext: context});
     }
     args = args.concat(message);
 
     var data = {
+        assertionFunction: callSpec.enhancedFunc,
         originalMessage: message,
+        defaultMessage: callSpec.matcherSpec.defaultMessage,
         enhanced: enhanced,
         args: args
```
Comment (jamestalmage) [inline lib/decorator.js:95]: but this is the event object, and it doesn't have it.

**Comment by twada** [comment]
> just attach their entire original specification to the event, instead of copying over individual properties. That way they can attach arbitrary data specific to their own handlers

Now I understand and it sounds good to me ;)

**Comment by jamestalmage** [comment]
OK, 
I've made all the changes and attached some documentation. We will need to document it better at some point, but I'd like to get this into use in AVA and make sure we like the API before spending a lot of time on that.

**Comment by twada** [comment]
Thanks and agreed. If you have any changes not pushed yet, would you update this pull-req?

**Comment by jamestalmage** [comment]
Sorry, I was working on a different branch and forgot.

**Comment by twada** [comment]
@jamestalmage looks good to me. Thank you for the great contribution!

**Comment by twada** [comment]
@jamestalmage Just released [empower-core 0.3.0](https://github.com/twada/empower-core/blob/master/CHANGELOG.md#030-2016-01-06) :tada: 

I really appreciate your contributions.

**Comment by jamestalmage** [comment]
Thanks @twada - I will write up an AVA PR using it in a few days. Thanks for working with us on making `power-assert` better for AVA.

**Pull request merged**